### ISSUE TITLE
Show unlisted projects from Giveth on DeVouch

### ISSUE THREAD

**Issue by divine-comedian** [Giveth/DeVouch-BE#154]
Reported by @GriffGreen a project supporter wanted to vouch for a new project on Giveth. However the project did not yet have "listed" status by the app. 

When navigating to the devouch page for the given project it threw an application error. The DeVouch back-end also holds no data for the project.

I listed the project in the admin and 10 minutes later the project page was shown correctly on DeVouch. We should handle this case on DeVouch and show projects that are not yet listed on Giveth. 

However we should NOT SHOW projects that have status of "deactivated" and "cancelled"

**Comment by divine-comedian** [comment]
this seems to be working as expected on staging, however on production it still shows projects that are deactivated 

this has been in QA since december last year and we've had at least 1 release I think since then. 

@kkatusic any reason you can think why this isn't working already as expected on production?

**Comment by kkatusic** [comment]
@divine-comedian on staging it is also showing projects from production. I can't find what project is showing on the devouch that has been deactivated on the giveth :(

But no problem, we have bigger problem if you deactivate some project on Giveth it will still be on the DeVouch. Also if you for some reason _vouch_ ot _flag_ Giveth's project you can delete it because of the foreign key relation inside sql database.

How BE works, it fetch all projects, 50 by 50..., after that it will update or add new ones, there is no option to delete project if it is deactivated or cancelled on Giveth's side.

**Comment by divine-comedian** [comment]
@kkatusic you can use some test project on Giveth production to do this or do you need production admin panel access? I can provide this as well

You can compare devouch production and staging behaviour for when you change project status on Giveth.

I noticed for me the behaviour was working as expected on staging, 

maybe it is not deleting the project on devouch but simply hiding it on the FE - this would have been the reasonable approach considering that projects can be reactivated.

 Did you look at any code previously implemented to see how it was handled?

**Comment by kkatusic** [comment]
@divine-comedian I done that, deactivated one project and waited, no changes, also checked code on the BE, That's why I wrote the it won't be deleted.

Code that is related to this issue only added option to list unlisted projects, both setup on devauch BE and giveth BE.

I think, not 100% sure, when you reset/re-deploy code on devouch staging BE it reset database

**Comment by kkatusic** [comment]
@divine-comedian to fix this issue we need to do first some changes on the Giveth BE GraphQL. We need query that will return to us also deactivated and/or canceled projects. If we don't have that kind of data like is now script will not know what project to "hide".

Second step would be to add "active" column to the projects table inside DeVouch database and update that value, only active should be listed.

How now it works, it fetch all active projects from the Giveth BE, check if that project exists, if not add new one or update existing one. No other logics inside functions for fetching projects.

One consideration, this way to fetch all projects from Giveth it runs inside this code:

https://github.com/Giveth/DeVouch-BE/blob/develop/src/features/import-projects/giveth/index.ts

this will loop all projects and make about approx. 70 requests when it started, 50 projects per query, something that we avoided  when building SEO sitemap projects list. This impact only Giveth BE, DeVouch seems to runs fast.

**Comment by divine-comedian** [comment]
@kkatusic 

got it, so we fetch all projects, check if ones that are deactivated and cancelled are in DB, mark to not return them to FE results 

how much extra workload is this for the BE compared to what we are running now?

**Comment by kkatusic** [comment]
@divine-comedian 

we need to provide new API endpoint for DeVouch on Giveth BE, should be easy, copy-paste existing or we can manage existing one to fetch all projects inside database, will check with BE dev.

Add one active field to the DV BE database projects table and add that flag on the FE side to not show non active projects.

**Comment by divine-comedian** [comment]
ah so it is a new endpoint on Giveth BE that provides only projects that meet our criteria. 

Do you need then a BE dev on Giveth to work on it?

**Comment by kkatusic** [comment]
We should only add additional filter to `allProjects` query, something like `fetchAll` and skip on that way to return only active projects. I will wait Carlos to finish his work then I will ask him for hotfix ;)

**Comment by kkatusic** [comment]
@laurenluz will we include this to the DeVouch? Be aware that we are reading projects from the V5 BE side via API and all these projects would be exposed to see if we adjust query for example with the flag: `showAll=1`

**Comment by laurenluz** [comment]
DeVouch should not treat all unlisted projects as eligible to show. Since unlisted can mean either temporarily pending review or intentionally hidden by an admin, DeVouch needs either live Giveth status sync plus a clearer visibility signal, or it should default to hiding unlisted projects

I don't think though that this is worth our time to build a live syncing solution since afaik there are only few users currently on devouch @kkatusic - wdyt?

**Comment by kkatusic** [comment]
@laurenluz I agree with you, we can close then this issue and reopen if we need this option in the future.
